x/incentives/keeper/gauge_test.go

```diff
@@ -1114,6 +1114,101 @@ func (s *KeeperTestSuite) TestChargeGroupCreationFeeIfNotWhitelisted() {
 	}
 }
 
+func (s *KeeperTestSuite) TestGetPoolIdsAndDurationsFromGroup() {
+	s.SetupTest()
+
+	pools := s.PrepareAllSupportedPools()
+
+	testCases := []struct {
+		name              string
+		expectedPoolIds   []uint64
+		expectedDurations []time.Duration
+	}{
+		{
+			name:              "Two pool IDs of various types",
+			expectedPoolIds:   []uint64{pools.BalancerPoolID, pools.StableSwapPoolID},
+			expectedDurations: []time.Duration{time.Hour * 7, time.Hour * 7},
+		},
+		{
+			name:              "Three pool IDs of various types",
+			expectedPoolIds:   []uint64{pools.ConcentratedPoolID, pools.BalancerPoolID, pools.StableSwapPoolID},
+			expectedDurations: []time.Duration{time.Hour * 24 * 7, time.Hour * 7, time.Hour * 7},
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			equalPoolVolumes, _ := setupEqualVolumeWeights(len(tc.expectedPoolIds), oneMillionVolumeAmt)
+			s.SetupVolumeForPools(tc.expectedPoolIds, equalPoolVolumes, map[uint64]osmomath.Int{})
+
+			gaugeId, err := s.App.IncentivesKeeper.CreateGroup(s.Ctx, defaultCoins, types.PerpetualNumEpochsPaidOver, s.TestAccs[0], tc.expectedPoolIds)
+			s.Require().NoError(err)
+
+			group, err := s.App.IncentivesKeeper.GetGroupByGaugeID(s.Ctx, gaugeId)
+			s.Require().NoError(err)
```

Comment:
Similarly, we don't need to create a group in state since this method doesn't end up reading the `Group` or group `Gauge` from state.

Instead, we can define `Group` on the test case so that we can provide the right configuration as an argument to the system under test - `tc.group`.

The reason I'm advocating for this is that when I see a `Group` created in-state via `CreateGroup`, I start thinking that the system under test relies on it being present in state. This is misleading and deters the reader from focusing on the actual properties the test is trying to validate.

Reply:
yes, good call on test refactors, done here as well https://github.com/osmosis-labs/osmosis/pull/6601/commits/cd2036cd27221e9232c2ce609b128cd8cd7de388